### PR TITLE
Start working on escaping entities' output to avoid injections

### PR DESCRIPTION
**Pull request by Pierstoval**

### CODE REVIEW

File: src/Glpi/Search/Provider/SQLProvider.php [hunk 1]
```diff
@@ -5140,7 +5140,7 @@ public static function giveItem(
                                                 $out .= \Search::LBBR;
                                             }
                                             $count_display++;
-                                            $out .= "<a href='mailto:" . $split2[1] . "'>" . $split2[1] . "</a>";
+                                            $out .= "<a href='mailto:" . $split2[1] . "'>" . \htmlspecialchars($split2[1]) . "</a>";
```
Comment (cedric-anne) [inline src/Glpi/Search/Provider/SQLProvider.php:5143]: Every value should be escaped in every piece of HTML code.

```suggestion
                                            $out .= "<a href='mailto:" . \htmlspecialchars($split2[1]) . "'>" . \htmlspecialchars($split2[1]) . "</a>";
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/CommonDBTM.php [hunk 2]
```diff
@@ -1420,7 +1420,7 @@ public function getLink($options = [])
 
         $link = $this->getLinkURL();
 
-        $label = $this->getNameID($options);
+        $label = \htmlspecialchars($this->getNameID($options));
```
Comment (cedric-anne) [inline src/CommonDBTM.php:1423]: This one is far more complex. `getNameID` may return a HTML snipper or a raw name, depending on paramater. Changes made on this function should be done in a dedicated PR, as it may be really complex to validate.
```suggestion
        $label = $this->getNameID($options);
```

File: src/Glpi/Features/PlanningEvent.php [hunk 3]
```diff
@@ -889,7 +889,7 @@ public function getAlreadyPlannedInformation(array $val)
                  $out .= "&amp;forcetab=" . $itemtype . "$1";
             }
             $out .= "'>";
-            $out .= Html::resume_text($val["name"], 80) . '</a>';
+            $out .= \htmlspecialchars(Html::resume_text($val["name"], 80)) . '</a>';
```
Comment (cedric-anne) [inline src/Glpi/Features/PlanningEvent.php:892]: `Html::resume_text()` returns a `&nbsp;`, therefore, the encoding should be done in this method directly, in a dedicated PR too, probably.

File: src/Glpi/Features/PlanningEvent.php [hunk 4]
```diff
@@ -1104,7 +1104,7 @@ public static function getSpecificValueToDisplay($field, $values, array $options
                     $users[] = sprintf(
                         '<a href="%s">%s</a>',
                         User::getFormURLWithID($user_id),
-                        getUserName($user_id, 1)
+                        \htmlspecialchars(getUserName($user_id, 1))
```
Comment (cedric-anne) [inline src/Glpi/Features/PlanningEvent.php:1107]: `getUserName()` is already returning a HTML snippet when the `$link` param is no 0.
```suggestion
                        getUserName($user_id, 1)
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]
